pkg/controller/controller_test.go

```diff
@@ -3876,6 +3925,206 @@ func TestDelete(t *testing.T) {
 			expectErr:  false,
 			mockDelete: true,
 		},
+		"Empty provisioner secret is set as PV annotation and StorageClass exists": {
+			persistentVolume: &v1.PersistentVolume{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "pv",
+					Annotations: map[string]string{
+						AnnDeletionProvisionerSecretRefName:      "",
+						AnnDeletionProvisionerSecretRefNamespace: "",
+					},
+				},
+				Spec: v1.PersistentVolumeSpec{
+					PersistentVolumeSource: v1.PersistentVolumeSource{
+						CSI: &v1.CSIPersistentVolumeSource{
+							VolumeHandle: "vol-id-1",
+						},
+					},
+				},
+			},
+			storageClass: &storagev1.StorageClass{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "sc-name",
+				},
+				Parameters: map[string]string{},
+			},
+			expectErr:                 false,
+			mockDelete:                true,
+			expectedProvisionerSecret: &expectedSecret{exist: false},
+		},
+		"Empty provisioner secret is set as PV annotation and StorageClass doesn't exist on deletion": {
+			persistentVolume: &v1.PersistentVolume{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "pv",
+					Annotations: map[string]string{
+						AnnDeletionProvisionerSecretRefName:      "",
+						AnnDeletionProvisionerSecretRefNamespace: "",
+					},
+				},
+				Spec: v1.PersistentVolumeSpec{
+					PersistentVolumeSource: v1.PersistentVolumeSource{
+						CSI: &v1.CSIPersistentVolumeSource{
+							VolumeHandle: "vol-id-1",
+						},
+					},
+				},
+			},
+			expectErr:                 false,
+			mockDelete:                true,
+			expectedProvisionerSecret: &expectedSecret{exist: false},
+		},
+		"Non-empty provisioner secret is set as PV annotation and StorageClass exists": {
+			persistentVolume: &v1.PersistentVolume{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "pv",
+					Annotations: map[string]string{
+						AnnDeletionProvisionerSecretRefName:      "provisionersecret",
+						AnnDeletionProvisionerSecretRefNamespace: defaultSecretNsName,
+					},
+				},
+				Spec: v1.PersistentVolumeSpec{
+					PersistentVolumeSource: v1.PersistentVolumeSource{
+						CSI: &v1.CSIPersistentVolumeSource{
+							VolumeHandle: "vol-id-1",
+						},
+					},
+				},
+			},
+			storageClass: &storagev1.StorageClass{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "sc-name",
+				},
+				Parameters: map[string]string{
+					prefixedProvisionerSecretNameKey:      "provisionersecret",
+					prefixedProvisionerSecretNamespaceKey: defaultSecretNsName,
```

Comment:
Use a different key-value par than `getDefaultProvisinerSecrets()` to see where the final secret comes from. This applies to all tests below.

Reply:
@jsafrane  Thank you for your review.

`Non-empty provisioner secret is set as PV annotation and modified StorageClass exists` and `Non-existent provisioner secret is set as PV annotation` already checks where the final secret comes from.
Former one has invalid secret specified in the StorageClass and verifies that the right secret is taken from annotations (we may still need to discuss if it should be the expected behavior as commented [here](https://github.com/kubernetes-csi/external-provisioner/pull/713#discussion_r829110566)).
Latter one has invalid secret specified in the annotations and verifies that it fails to get the secret. 

However, we may be able to improve `getDefaultProvisinerSecrets()` to prepare two secrets and check that either of the expected secret is set, to make it clear.
(Also, `getDefaultProvisinerSecrets()` should be renamed to something like `getProvisinerSecretCandidates()`?)

Reply:
@jsafrane could you please confirm/share your thought on above , so that the unit test can also be adjusted..